### PR TITLE
Create a tool-test-helper action

### PR DESCRIPTION
**Pull request by laurit17**
Analogous to the linter one

### CODE REVIEW

File: .trunk/trunk.yaml [hunk 1]
```diff
@@ -42,14 +42,23 @@ actions:
         - git_hooks: [pre-push]
 
     - id: linter-test-helper
-      display_name: Test Generator
+      display_name: Linter Test Generator
       description: Generate boilerplate test code when new linters are defined
```
Comment (TylerJang27) [inline .trunk/trunk.yaml:46]: Do we want the linter boilerplate to now include tools?

File: repo-tools/tool-test-helper/generate [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import os
+from datetime import datetime, timedelta
+
+import click
+
+INITIAL_PLUGIN_CONTENTS = """version: 0.1
+downloads:
+  - name: NAME_HERE
+    # executable: true
+    # NOTE: These are (common) sample values. Please replace with real values.
```
Comment (TylerJang27) [inline repo-tools/tool-test-helper/generate:12]: Maybe mention something about how they're a mapping and can be resolved for ${version} ${os} ${cpu}

File: repo-tools/tool-test-helper/generate [hunk 3]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import os
+from datetime import datetime, timedelta
+
+import click
+
+INITIAL_PLUGIN_CONTENTS = """version: 0.1
+downloads:
```
Comment (TylerJang27) [inline repo-tools/tool-test-helper/generate:9]: Maybe mention that a tool can have a download or a runtime. The linter generate script doesn't include a download section (maybe it should?)

File: repo-tools/tool-test-helper/generate [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    downloads:
+      - os:
+          linux: linux
+          macos: darwin
+          windows: windows
+        cpu:
+          amd64: amd64
+          arm64: arm64
+        url: https://URL_HERE
+lint:
```
Comment (TylerJang27) [inline repo-tools/tool-test-helper/generate:22]: tools?

File: repo-tools/tool-test-helper/generate [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  definitions:
+    - name: NAME_HERE
+      # RUNTIME_OR_DOWNLOAD_INFO_HERE
+      # download: NAME_HERE
+      known_good_version: # VERSION_HERE
```
Comment (TylerJang27) [inline repo-tools/tool-test-helper/generate:27]: standardize when fields are comment vs. required. Suggestion:
`known_good_version: VERSION_HERE`

File: .trunk/trunk.yaml (hunk 1 above)
Comment (laurit17) [reply .trunk/trunk.yaml:46]: done, good idea

**Review by TylerJang27 — APPROVED** [review]
LGTM

**Pull request merged**